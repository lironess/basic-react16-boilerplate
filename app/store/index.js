import { createStore, applyMiddleware, compose } from 'redux';
import { root } from 'reducers/root';

const middlewares = [];

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const store = createStore(root, composeEnhancers(applyMiddleware(...middlewares)));

export default store;
