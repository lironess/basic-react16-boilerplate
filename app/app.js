import React from 'react';
import { Provider, connect } from 'react-redux';
import PropTypes from 'prop-types';

import store from 'store/index';

const Counter = ({ count }) => <div>{ count }</div>;
const ConnectedCounter = connect(({ count }) => ({ count }))(Counter);

export const App = () => (
  <Provider store={ store }>
    <div>
      Hello world
      <ConnectedCounter />
    </div>
  </Provider>
);

Counter.propTypes = {
  count: PropTypes.number
};

Counter.defaultProps = {
  count: 0
};
