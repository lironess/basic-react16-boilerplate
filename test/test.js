import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { App } from '../app/app';

Enzyme.configure({ adapter: new Adapter() });

test('App should render', () => {
  const component = shallow(<App />);

  expect(component).toMatchSnapshot();
});
