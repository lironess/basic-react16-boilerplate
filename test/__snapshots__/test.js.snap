// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <div>
    Hello world
    <Connect(Counter) />
  </div>
</Provider>
`;
